src/components/fileviewer.js

```diff
@@ -16,15 +17,11 @@ const _ = require('lodash');
 export default class FileViewerContainer extends Component {
   constructor(props) {
     super(props);
+
+    /* app status */
+    this.status = new FileViewerStatus();
```

Comment:
This is the important change: The `FileViewerStatus` is a `Component` inside the `FileViewerContainer`, and it has it's own state